types/time.go

```diff
@@ -649,6 +655,16 @@ func parseDatetime(str string, fsp int, isFloat bool) (Time, error) {
 				fmt.Sscanf(fracStr, "%2d%2d%2d", &hour, &minute, &second)
 			}
 		}
+		if l == 9 || l == 10 {
+			switch len(fracStr) {
+			case 0:
+				second = 0
+			case 1:
+				_, err = fmt.Sscanf(fracStr, "%1d", &second)
+			default:
+				_, err = fmt.Sscanf(fracStr[:2], "%2d", &second)
+			}
```

Comment:
should we raise warning as MySQL does for the case below?
```
mysql> select cast("170102037.1122" as datetime);
+------------------------------------+
| cast("170102037.1122" as datetime) |
+------------------------------------+
| 2017-01-02 03:07:11                |
+------------------------------------+
1 row in set, 1 warning (0.00 sec)

mysql> show warnings;
+---------+------+------------------------------------------------------+
| Level   | Code | Message                                              |
+---------+------+------------------------------------------------------+
| Warning | 1292 | Truncated incorrect datetime value: '170102037.1122' |
+---------+------+------------------------------------------------------+
1 row in set (0.00 sec)
```

Reply:
Yes, indeed. I'll fix here